starlark/tests/rust-testcases/dict.sky

```diff
@@ -0,0 +1,3 @@
+# Dict tests
+
+{[]: 1 for x in [1]}    ### Value is not hashable
```

Comment:
Why not:
```suggestion
{[]: 1}    ### Value is not hashable
```
?

Reply:
`{[]: 1}` correctly returns `Value is not hashable` in master, panic only happens in dict comprehensions.

Reply:
Ok make sense it is probably happening deeper in the evaluation graph.